resources/test_fonts/ttx/names_only.ttx

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ttFont sfntVersion="\x00\x01\x00\x00" ttLibVersion="4.37">
+  <name>
+    <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
+      NameTest
+    </namerecord>
+    <namerecord nameID="2" platformID="1" platEncID="0" langID="0x0">
+      Regular
+    </namerecord>
+    <namerecord nameID="2" platformID="3" platEncID="1" langID="0x408">
+      Κανονικά
+    </namerecord>
+    <namerecord nameID="2" platformID="3" platEncID="1" langID="0x415">
+      Normalny
+    </namerecord>
+    <namerecord nameID="2" platformID="3" platEncID="1" langID="0x42d">
+      Arrunta
+    </namerecord>
+    <namerecord nameID="2" platformID="3" platEncID="1" langID="0x804">
+      正常
```

Comment:
Thanks for adding that, one thing that came to my mind after the review yesterday: IIRC we've had bugs in Chromium where we didn't handle the encoding correctly, or the font had the wrong encoding. I tried finding the bug, but what this raised: To what extent do we want to test how we handle broken fonts here? What happens if there's invalid UTF-8 in the font? (I assume the records are stored in UTF-8?)

Reply:
Unicode names are encoded in UTF-16BE. The decoding is handled here: https://github.com/googlefonts/fontations/blob/e3dc30dca577cee306c4256d0ef8c1e31fc88dfe/read-fonts/src/tables/name.rs#L147

Invalid characters are emitted as U+FFFD (Unicode replacement character) so it’s a potentially lossy conversion.

Reply:
It looks like if the buffer ends with the first half of a surrogate pair, it is simply dropped. We probably want to emit a replacement character in that case as well.